File: GameOFLife/app/src/main/java/com/ludeman/gameoflife/SquareCell.java
package com.ludeman.gameoflife;

import java.util.List;

public class SquareCell {
    final Location cellLocation;
    State cellState;
    List<SquareCell> neighbors;

    public SquareCell(Location location, Integer alive) {
        cellLocation = location;
        if (alive == 1) cellState = State.ALIVE;
        else cellState = State.DEAD;
    }

    public void died(){
        cellState = State.DEAD;
    }

    public void revived() {
        cellState = State.ALIVE;
    }
}

File: GameOFLife/app/src/main/java/com/talinapps/gameoflife/SquareCell.kt
package com.talinapps.gameoflife

class SquareCell(val cellLocation: Location, alive: Int) {
    var cellState: State = if (alive == 1) State.ALIVE else State.DEAD
    var neighbors: List<SquareCell> = listOf()

    fun died() {
        cellState = State.DEAD
    }

    fun revived() {
        cellState = State.ALIVE
    }
}
